src/graphql/data/nft/Collection.ts

```diff
@@ -0,0 +1,132 @@
+            }
+            listings
+            volume(duration: DAY) {
+              value
+              currency
+            }
+            volumePercentChange(duration: DAY) {
+              value
+              currency
+            }
+            floorPricePercentChange(duration: DAY) {
+              value
+              currency
+            }
+          }
+        }
+      }
+      pageInfo {
+        endCursor
+        hasNextPage
+        hasPreviousPage
+        startCursor
+      }
+    }
+  }
+`
+export function useCollectionQuery(address: string): GenieCollection | undefined {
+  const collectionQueryReference = loadQuery<CollectionQuery>(RelayEnvironment, collectionQuery, {
+    address,
+  })
+  const queryData = usePreloadedQuery<CollectionQuery>(collectionQuery, collectionQueryReference)
+
+  const queryCollection = queryData.nftCollections?.edges[0].node
+  const traits = {} as Record<string, Trait[]>
+  if (queryCollection?.traits) {
+    for (const trait of queryCollection?.traits) {
+      traits[trait.name ?? ''] =
+        trait.values?.map((value) => {
+          return {
+            trait_type: trait.name,
+            trait_value: value,
+          } as Trait
+        }) ?? ({} as Trait[])
+    }
+  }
+  return {
+    address,
+    isVerified: queryCollection?.isVerified ?? undefined,
+    name: queryCollection?.name ?? undefined,
+    description: queryCollection?.description ?? undefined,
+    standard: queryCollection?.nftContracts ? queryCollection?.nftContracts[0].standard ?? undefined : undefined,
+    bannerImageUrl: queryCollection?.bannerImage?.url ?? undefined,
+    stats: queryCollection?.markets
+      ? {
+          num_owners: queryCollection?.markets[0].owners ?? undefined,
+          floor_price: queryCollection?.markets[0].floorPrice?.value ?? undefined,
+          one_day_volume: queryCollection?.markets[0].volume?.value ?? undefined,
+          one_day_change: queryCollection?.markets[0].volumePercentChange?.value ?? undefined,
+          one_day_floor_change: queryCollection?.markets[0].floorPricePercentChange?.value ?? undefined,
+          banner_image_url: queryCollection?.bannerImage?.url ?? undefined,
+          total_supply: queryCollection?.numAssets ?? undefined,
+          total_listings: queryCollection?.markets[0].listings ?? undefined,
+          total_volume: queryCollection?.markets[0].totalVolume?.value ?? undefined,
+        }
+      : undefined,
+    traits,
+    // marketplaceCount: { marketplace: string; count: number }[], // TODO add when backend supports
+    imageUrl: queryCollection?.image?.url ?? undefined,
+    twitter: queryCollection?.twitterName ?? undefined,
+    instagram: queryCollection?.instagramName ?? undefined,
+    discordUrl: queryCollection?.discordUrl ?? undefined,
+    externalUrl: queryCollection?.homepageUrl ?? undefined,
+    rarityVerified: false, // TODO update when backend supports
+    // isFoundation: boolean, // TODO ask backend to add
```

Comment:
the amount of nested conditional data mapping and fallbacks feels messy here--indicates to me that there should be a cleaner approach, or that the consumer/upstream data source should change. (i.e., mapping `queryCollection.twitterName` to `twitter` smells like a data problem that we shouldn't allow to continue.

Any thoughts?

Reply:
For context, when deciding on doing this migration, we saw that there was a discrepancy between how data team wanted to store data moving forward and what was done previously/what the components have been built around. The compromise agreed upon was for fast delivery but allowing for the migration of the most data intensive calls was to create these mapping functions that map the graphql data to the types currently used in the app. This will be the interim solution for phase1 and plan to migrate to objects that more closely align with the graphql schemas in the future. Sent comments in slack for reasons behind the null check logic and how it can be remediated.